### PR TITLE
Implement wasmJs support

### LINKED ISSUE #16: WasmJS target
Would be great to provide a wasmJs target.

### PR DESCRIPTION
**Pull request by stashymane**
WASM code mostly copied from desktop/JVM target, with the addition of an `IODispatcher` common val - `Dispatchers.IO` does not exist on WASM and is replaced with the default dispatcher. Since in the majority of cases this dispatcher will be used for network calls or file fetching, which is done completely asynchronously on wasmJs anyway, performance is actually relatively good on the web. The only downside currently is that `getProcessorCount` is hardcoded to return 8 on wasm, as there's no good way to figure out how many concurrent connections the browser can support. A better approach would be to allow the library user to define how many threads to allocate, or provide a coroutine scope themselves.

Closes #16

### CODE REVIEW

File: library/src/wasmJsMain/kotlin/ovh/plrapps/mapcompose/utils/IODIspatcher.android.kt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package ovh.plrapps.mapcompose.utils
+
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+
+actual val IODispatcher: CoroutineDispatcher = Dispatchers.Default
```
Comment (p-lr) [inline library/src/wasmJsMain/kotlin/ovh/plrapps/mapcompose/utils/IODIspatcher.android.kt:6]: I believe this file should be renamed `IODispatcher.wasm.kt`

File: library/src/commonMain/kotlin/ovh/plrapps/mapcompose/utils/IODIspatcher.kt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package ovh.plrapps.mapcompose.utils
```
Comment (p-lr) [inline library/src/commonMain/kotlin/ovh/plrapps/mapcompose/utils/IODIspatcher.kt:1]: Not very important, but this file and the others should be renamed `IODispatcher...`

File: library/src/wasmJsMain/kotlin/ovh/plrapps/mapcompose/ui/state/MapState.wasm.kt [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package ovh.plrapps.mapcompose.ui.state
+
+actual fun getProcessorCount(): Int {
+    return 8
```
Comment (p-lr) [inline library/src/wasmJsMain/kotlin/ovh/plrapps/mapcompose/ui/state/MapState.wasm.kt:4]: Perhaps adding a comment here (like what you explained in the PR description) would be nice here.

**Review by p-lr — COMMENTED** [review]
Thanks ! Really nice work here. I also appreciate your detailed description.

**Comment by atesahmet0** [comment]
Thanks for this feature 🧿

**Pull request merged**